### ISSUE TITLE
Unknown column type "double" requested

### ISSUE THREAD

**Issue by mabehiry** [rainlab/builder-plugin#295]
I get this error when clicking  save button:
Unknown column type "double" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgot to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information.

This is my code:
```php
public function up()
{
    Schema::table('sunsoft_software_order_items', function($table)
    {
        $table->integer('item_id')->default(null)->change();
        $table->double('price', 10, 0)->default(null)->change();
        $table->string('details', 191)->default(null)->change();
        $table->double('quantity', 10, 0)->default(null)->change();
        $table->double('total', 10, 0)->default(null)->change();
        $table->timestamp('created_at')->default(null)->change();
        $table->timestamp('updated_at')->default(null)->change();
        $table->timestamp('deleted_at')->default(null)->change();
        $table->dropColumn('item');
    });
}

public function down()
{
    Schema::table('sunsoft_software_order_items', function($table)
    {
        $table->integer('item_id')->default(NULL)->change();
        $table->double('price', 10, 0)->default(NULL)->change();
        $table->string('details', 191)->default('NULL')->change();
        $table->double('quantity', 10, 0)->default(NULL)->change();
        $table->double('total', 10, 0)->default(NULL)->change();
        $table->timestamp('created_at')->default('NULL')->change();
        $table->timestamp('updated_at')->default('NULL')->change();
        $table->timestamp('deleted_at')->default('NULL')->change();
        $table->integer('item')->nullable()->default(NULL);
    });
}
```

![Screenshot_2](https://user-images.githubusercontent.com/33180080/63229981-41cbd780-c224-11e9-850f-3e05f61b664f.png)

**Comment by bennothommo** [comment]
@mabehiry thanks for the report. Would you mind letting me know which type of database server are you running?

**Comment by mabehiry** [comment]
Hello,
Database server is:
MariaDB 10.3.16
Thank you

**Comment by bennothommo** [comment]
@mabehiry I have run the same migration on a test server with MariaDB 10.3 and did not encounter any issues. Would you be able to confirm which version of October you are running, and which version of `doctrine/dbal` is installed in Composer if you installed that way? Also, please confirm that you have not got any plugins or additions manipulating the available column types in Doctrine\DBAL.

**Comment by mabehiry** [comment]
Hello,
I'm running on build 455, I didn't install doctrine/dbal manually. I'm running  on local machine with xampp server. there are no plugins installed except those:
Uploader - Builder - Blog + my  plugin
Also, default value for most columns  are set automatically to NULL as shown  here

![image](https://user-images.githubusercontent.com/33180080/63590957-100f9380-c5cb-11e9-9726-42f41a564d7f.png)

When I click save without any update, I get this error:
![image](https://user-images.githubusercontent.com/33180080/63590986-261d5400-c5cb-11e9-8d64-1d9b34e3f846.png)

When clicking ok, it  continues loading:
![image](https://user-images.githubusercontent.com/33180080/63591009-333a4300-c5cb-11e9-955f-624a2c9e1cfc.png)

When I remove the null value and save, the next time it get displayed again.

This is my environment:
XAMPP 3.2.4
Apache/2.4.39 (Win64)
Database client version: libmysql - mysqlnd 5.0.12-dev
Server type: MariaDB
Server version: 10.3.16-MariaDB 
PHP version: 7.3.8

**Comment by janvandelaar** [comment]
I have the same problem, also on a local machine, using Xampp 3.2.4
When I create a field named 'created_at' and choose Date and Time as a type, it gets saved as 'Timestamp'. This way I can't use the reorder functions for my  backend lists.

**Comment by faselunare** [comment]
Hi same problem here with timestamp type
i have created a simple plugin with the builder (just for test this issue that i have discovered in one of my plugins)
``` 
public function up()
{
    Schema::table('test_test_events', function($table)
    {
        $table->timestamp('created_at')->nullable()->change();
        $table->timestamp('updated_at')->nullable()->change();
        $table->timestamp('deleted_at')->nullable()->change();
        $table->string('title', 191)->nullable()->change();
    });
}

public function down()
{
    Schema::table('test_test_events', function($table)
    {
        $table->timestamp('created_at')->default('NULL')->change();
        $table->timestamp('updated_at')->default('NULL')->change();
        $table->timestamp('deleted_at')->default('NULL')->change();
        $table->string('title', 191)->default('NULL')->change();
    });
}
```
when i try to modify the table (adding a new integer field         $table->integer('weight')->nullable(); )
i get the doctrine error

i'm on October 460
Mysql Ver 15.1 Distrib 10.3.17-MariaDB, for debian-linux-gnueabihf (armv7l) using readline 5.2
PHP 7.3.11

**Comment by albanoalfredo** [comment]
Having the same issue with a timestamp.  Where you able to solve it?
PHP version: 7.3.6
Server version: 10.3.21-MariaDB - MariaDB Server

**Comment by Athov** [comment]
I have the same issue it comes from the default value been set to 'null' instead of null (adding quotes) . The funny thing is that I have multiple octobers and only on few of them (the newest) this happen. All octobers are version 464 but the new are upgraded from 454 and the old are from 43x(maybe 434 not sure).
![image](https://user-images.githubusercontent.com/8526043/73522213-7f0dd580-4410-11ea-811a-a549f263fd22.png)

**Comment by bennothommo** [comment]
@faselunare @albanoalfredo Are both of you using XAMPP for development? The only common thread I've seen so far from all reports on this issue have been that people are using XAMPP with MariaDB.

**Comment by bennothommo** [comment]
@Athov would you be able to check in a different program what the default value for those fields are? `'null'` (with quotes) is only used to define "null" as a string as a default value for a field. It was done as part of the changes here - https://github.com/rainlab/builder-plugin/pull/291. It's possible if you had entered `NULL` in an older version of the Builder plugin (ie. before 1.0.26), those tables created would have their default values set to the string "NULL", not a true NULL value.

**Comment by cosminbosutar** [comment]
Hi,
I have the same problem as in the title, when I try to update a table that contains a double i receive the same error message popup.
I use laradock with nginx and mysql for local development:
  PHP 7.3.13
  Server type: MySQL
  Server version: 8.0.19 - MySQL Community Server - GPL

**Comment by albanoalfredo** [comment]
Hi,
I'm not using xampp.I have it installed on two separated servers, it only happens on the one that has mariaDB.I was able to "solve" it by cleaning up the migration table when I save the table.

I have to delete all the additional lines it tries to create when updating (selected on the screenshot below)

![image](https://user-images.githubusercontent.com/38056345/74885582-d77c2700-536d-11ea-9241-a77b140ab847.png)

**Comment by bennothommo** [comment]
@cosminbosutar @albanoalfredo Thanks for the further reports. I will investigate this further and get back to you.

**Comment by Athov** [comment]
> 
> 
> @Athov would you be able to check in a different program what the default value for those fields are? `'null'` (with quotes) is only used to define "null" as a string as a default value for a field. It was done as part of the changes here - #291.
> 
> It's possible if you had entered `NULL` in an older version of the Builder plugin (ie. before 1.0.26), those tables created would have their default values set to the string "NULL", not a true NULL value.

No All values are true NULL and not text in the newer version it tries to set the default('null') when trying to UPDATE the tables but I always have to manually clean it.

we are not using xampp or anything of the sort it's a linux server with nginx, php and mariadb, but the problem is not from them because for the dev server we have the same nginx,php and mariadb running for multiple projects and some have the problem and some don't.

**Comment by bennothommo** [comment]
@Athov does your operating system on your dev servers use different quotation mark symbols? That's probably the last thing I can think of that might be causing this.

**Comment by Athov** [comment]
> 
> 
> @Athov does your operating system on your dev servers use different quotation mark symbols? That's probably the last thing I can think of that might be causing this.

I think I found the problem we moved the "bug free" the project to another server running mariadb and it happen to it as well so it's maybe mariadb or something on the server
the OS is Debian

**Comment by faselunare** [comment]
@bennothommo no Xampp
just php and maria db installed on top of Raspbian (all softwre updated to the latest version)

**Comment by pittss** [comment]
Hi, try this plugin https://octobercms.com/plugin/cubecamp-doctrine , may be help you

**Comment by LukeTowers** [comment]
@bennothommo can we use what that plugin does to fix the issue in the builder plugin itself?

**Comment by bennothommo** [comment]
@LukeTowers looks like it just brings in another Composer library to add support and registers a provider.

@pittss Would you have any problems if we merge this in with the Builder plugin?

**Comment by pittss** [comment]
@bennothommo No problem. Once the problem is solved, I'll remove the plugin from the marketplace.

**Comment by bennothommo** [comment]
Hi all,

I have released some changes as part of #329 in order to fix the above issues. Would I be able to ask people to test these out on their development instances of October and let me know if it resolves the issues for you?

You may need to check out the plugin via Composer in order to test this out.

**Comment by daftspunk** [comment]
Fixed in v1.1.1
